Collection developer guide - BOTMETA (#67538)
ansible/ansible: docs/docsite/rst/dev_guide/developing_collections.rst

See the `content_collector README <https://github.com/ansible/content_collector>`_ for full details and usage guidelines.

BOTMETA.yml
-----------

The `BOTMETA.yml <https://github.com/ansible/ansible/blob/devel/.github/BOTMETA.yml>`_ is the source of truth for:
* ansibullbot
* the docs build for collections-based modules

Ansibulbot will know how to redirect existing issues and PRs to the new repo.
The build process for docs.ansible.com will know where to find the module docs.

.. code-block:: yaml

   $modules/monitoring/grafana/grafana_plugin.py:
       migrated_to: community.grafana
   $modules/monitoring/grafana/grafana_dashboard.py:
       migrated_to: community.grafana
   $modules/monitoring/grafana/grafana_datasource.py:
       migrated_to: community.grafana
   $plugins/callback/grafana_annotations.py:
       maintainers: $team_grafana
       labels: monitoring grafana
       migrated_to: community.grafana
   $plugins/doc_fragments/grafana.py:
       maintainers: $team_grafana
       labels: monitoring grafana
       migrated_to: community.grafana

`Example PR <https://github.com/ansible/ansible/pull/66981/files>`_

> * The ``migrated_to:`` key must be added explicitly for every *file*. You cannot add ``migrated_to`` at the directory level.

Reviewer: This is not technically true. The bot does support directory level keys. If you want to make this the convention, you should explain what external systems will be using this and why those require individual files instead of dirs.

Author: It says above

The `BOTMETA.yml <https://github.com/ansible/ansible/blob/devel/.github/BOTMETA.yml>`_ is the source of truth for:
* ansibullbot
* the docs build for collections-based modules

Author: Added `This is to allow module and plugin webdocs to be redirected to the new collection docs.`

* ``migrated_to:`` MUST be added for every:

  * module
  * plugin
  * module_utils
  * contrib/inventory script

* You do NOT need to add ``migrated_to`` for:

> 

Reviewer: files that never existed in core

Author: Good spot, added

> 

Reviewer: files that never existed in core

See the `content_collector README <https://github.com/ansible/content_collector>`_ for full details and usage guidelines.

BOTMETA.yml
-----------

The `BOTMETA.yml <https://github.com/ansible/ansible/blob/devel/.github/BOTMETA.yml>`_ is the source of truth for:
* ansibullbot

> * Docs build.

Reviewer 2: ```suggestion
* the docs build for collections-based modules
```

> Ansibulbot will know how to redirect existing issues and PRs to the new repo

Reviewer 2: ```suggestion
Ansibulbot will know how to redirect existing issues and PRs to the new repo.
```

The build process for docs.ansible.com will know where to find the module docs.

.. code-block:: yaml
      $modules/monitoring/grafana/grafana_plugin.py:
          migrated_to: community.grafana
      $modules/monitoring/grafana/grafana_dashboard.py:
          migrated_to: community.grafana
      $modules/monitoring/grafana/grafana_datasource.py:
          migrated_to: community.grafana
      $plugins/callback/grafana_annotations.py:
        maintainers: $team_grafana
        labels: monitoring grafana
        migrated_to: community.grafana
      $plugins/doc_fragments/grafana.py:
        maintainers: $team_grafana
        labels: monitoring grafana
        migrated_to: community.grafana

`Example PR <https://github.com/ansible/ansible/pull/66981/files>`_

> * The ``migrated_to:`` key must be added explicitly for every *file*, not directories

Reviewer 2: ```suggestion
* The ``migrated_to:`` key must be added explicitly for every *file*. You cannot add ``migrated_to`` at the directory level.
```

Author: ```suggestion
The ``migrated_to:`` key must be added explicitly for every *file*. You MUST add ``migrated_to`` at the directory level.
```

> * ``migrated_to:`` MUST be added for every file which is a:

Reviewer 2: ```suggestion
* ``migrated_to:`` MUST be added for every:
```

>   * modules

Reviewer 2: ```suggestion
  * module
```

  * plugin
  * module_utils
  * contrib/inventory script

> * It is NOT needed for:

Reviewer 2: ```suggestion
* You do NOT need to add ``migrated_to`` for:
```

  * Unit tests
  * Integration tests

>   * Docs, such as ``docs/docsite/rst/user_guide/``

Reviewer 2: ```suggestion
  * ReStructured Text docs (anything under ``docs/docsite/rst/``)
```
